hoot-core/src/main/cpp/hoot/core/io/OsmGbdxXmlWriter.h

```diff
@@ -0,0 +1,148 @@
+/*
+ * This file is part of Hootenanny.
+ *
+ * Hootenanny is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2015, 2016, 2017, 2018 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+#ifndef OSMGBDXXMLWRITER_H
+#define OSMGBDXXMLWRITER_H
+
+// hoot
+#include <hoot/core/io/PartialOsmMapWriter.h>
+
+// Boost
+#include <boost/shared_ptr.hpp>
+
+// Qt
+#include <QDir>
+#include <QFile>
+#include <QHash>
+#include <QXmlDefaultHandler>
+class QString;
+class QXmlStreamWriter;
+
+// Standard
+#include <deque>
+
+// geos
+#include <geos/geom/Envelope.h>
+
+namespace hoot
+{
+
+/**
+ * Writes an OsmMap to a GBDX XML format file.
+ */
+class OsmGbdxXmlWriter : public QXmlDefaultHandler, public PartialOsmMapWriter
+{
+public:
+
+  static std::string className() { return "hoot::OsmGbdxXmlWriter"; }
+
+  static unsigned int logWarnCount;
+
+  OsmGbdxXmlWriter();
+  virtual ~OsmGbdxXmlWriter();
+
+  virtual bool isSupported(QString url) { return url.toLower().endsWith(".gxml"); }
+
+  virtual QString supportedFormats() {return ".gxml";}
+
+  virtual void open(QString url);
+
+  void close();
+
+  void setPrecision(int p) { _precision = p; }
```

Comment:
This deserves a comment describing what p is. The number of total digits? The number of places after the decimal? Spell it out for dummies like me ;)

Reply:
Done